### DISCUSSION TITLE
Can I submit test orders? If not, can I test on a real shop?

### DISCUSSION THREAD

**Discussion by TWAH** [etsy/open-api#950] [Q&A] [answered]
Hi,

As the title says, I need to put through some test orders for my app. How is this done? I Can't find any info on it.

If it's not possible, can I connect my app in development to a real shop? My app only requires read permissions and my partner has an active Etsy shop.

I know that there used to be an endpoint where you could add a shop as a test shop for this purpose but I cant find it.

Thanks

Tom

**Comment by Gareth064** [comment]
You can only test against a real shop.

I test against my wife's shop, which brings in enough to pay the mortgage. But it's risky to do so. If you do anything to break terms of service, there is a chance connected shops (like my wife's) could be penalised and shutdown.

You can create your own shop, but to test reading orders from it, you need sales. You could put up a couple of listings for a penny and buy them. That will give you some test data maybe.

But please carefully read Etsy's terms of service around this subject. Here is a useful resource to get you started.

https://www.etsy.com/uk/legal/policy/api-testing-policy/169130941112

**Reply by PMStanley** [reply, marked as answer]
@Gareth064 has you covered, you absolutely should be working against a test shop in developer mode at the start.  When you're in a position to test against a live shop you can add a small number of provisional users to your app (note: this allegedly only works whilst your app is not approved for commercial access)

Send a POST request to https://openapi.etsy.com/v3/application/provisional-users/{user_id}
Your API key needs to be in the header via x-api-key

A successful call results in a 201 and the users user_id and login_name

Calling the same endpoint with a DELETE command removes the user.

Call https://openapi.etsy.com/v3/application/provisional-users with GET to see a list of users (note: your dev account will not be included so if only you have access you'll get an empty list returned).

**Reply by showtroylove** [reply]
@PMStanley I couldn't find the endpoint you described in your post.  Can you share the source for the documentation that describes this endpoint?

**Reply by PMStanley** [reply]
I found it when I went to submit my app for commercial usage, cannot for the life of me remember where I saw it as it was very well hidden and to be honest I could have done with knowing about it months before.

It's described [here](https://groups.google.com/g/etsy-api-v2/c/_TA7DrM2uSU/m/rSs2INR9orQJ), can confirm it worked for me as-of late last year.

**Reply by TWAH** [reply]
@PMStanley Thanks for that link for provisional users, that's exactly what I was looking for :)
